### PR TITLE
[Qt-Advanced-Docking-System] update to 3.8.2

### PR DESCRIPTION
**Pull request by vincent-nv**
**Describe the pull request**

- #### What does your PR fix?
Upgrade Qt-Advanced-Docking-System to 3.8.2

- #### Which triplets are supported/not supported? Have you updated the [CI baseline]
all, No

- #### Does your PR follow the [maintainer guide]
- Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
Yes

### CODE REVIEW

File: versions/baseline.json [hunk 1]
```diff
@@ -7717,4 +7717,4 @@
       "port-version": 0
     }
   }
-}
+}
```
Comment (Thomas1664) [inline versions/baseline.json:7720]: Please add the final newlines again.

File: ports/qt-advanced-docking-system/portfile.cmake [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
-        hardcode_version.patch
         config_changes.patch
 )
 
 vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
     OPTIONS 
         -DBUILD_EXAMPLES=OFF
-        -DVERSION_SHORT=3.8.0
+        -DADS_VERSION=3.8.2
```
Comment (JonLiu1993) [inline ports/qt-advanced-docking-system/portfile.cmake:15]: @vincent-nv, Could you please clarify why change the option VERSION_SHORT to ADS_VERSION? I can not find ADS_VERSION in the Upstream

Comment (vincent-nv) [reply ports/qt-advanced-docking-system/portfile.cmake:15]: CMakeLists.txt change in project  Qt-Advanced-Docking-System
https://github.com/githubuser0xFFFF/Qt-Advanced-Docking-System/blob/3.8.2/CMakeLists.txt

Before a patch was apply removing getting VERSION_SHORT  from git, now is correctly handled by Qt-Advanced-Docking-System but need to pass ADS_VERSION

CMakeLists excerpt for 3.8.2 : 
```
if(NOT ADS_VERSION)
    set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules ${CMAKE_MODULE_PATH})
    include(GetGitRevisionDescription)
    git_describe(GitTagVersion --tags)
    string(REGEX REPLACE "^([0-9]+)\\..*" "\\1" VERSION_MAJOR "${GitTagVersion}")
    string(REGEX REPLACE "^[0-9]+\\.([0-9]+).*" "\\1" VERSION_MINOR "${GitTagVersion}")
    string(REGEX REPLACE "^[0-9]+\\.[0-9]+\\.([0-9]+).*" "\\1" VERSION_PATCH "${GitTagVersion}")
    set(VERSION_SHORT "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}")
else()
    string(REGEX MATCHALL "[\.]" VERSION_DOT_MATCHES ${ADS_VERSION})
    list(LENGTH VERSION_DOT_MATCHES VERSION_DOT_COUNT)
    if(VERSION_DOT_COUNT EQUAL 2)
        set(VERSION_SHORT ${ADS_VERSION})
    else()
        message(FATAL_ERROR "ADS_VERSION must be in major.minor.patch format, e.g. 3.8.1. Got ${ADS_VERSION}")
    endif()
endif()
```

Comment (JonLiu1993) [reply ports/qt-advanced-docking-system/portfile.cmake:15]: Thanks for the clarification

**Review by JonLiu1993 — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**